readme.md

```diff
@@ -1,54 +1,116 @@
-and utilities for creating Feign clients in commonly used configurations. Additionally,
-offers a basic round-robin failover client configuration for basic failover between multiple
-equivalent endpoints.
 
-- **retrofit-clients, retrofit2-clients**: Similar to `http-clients`, but generates proxies using the
-  [Retrofit](http://square.github.io/retrofit/) library.
+Core libraries:
+- jaxrs-clients: Clients for JAX-RS-defined service interfaces
+- retrofit2-clients: Clients for Retrofit-defined service interfaces
+- dropwizard-servers: Configuration library for Dropwizard/Jersey servers
 
-- **service-config**: Provides classes for setting up service clients from file-based or programmatic configuration.
+# Usage
 
-- **ssl-config**: Provides utilities for interacting with Java trust stores and key stores and acquiring
-  `SSLSocketFactory` instances using those stores, as well as a configuration class for use in server configuration
-  files.
+Maven artifacts are published to JCenter. Example Gradle dependency configuration:
 
-- **error-handling**: Provides utilities for relaying Java exceptions across JVM boundaries by serializing exceptions
-  as JSON POJOs.
+```groovy
+repositories {
+  jcenter()
+}
+
+dependencies {
+  compile "com.palantir.remoting1:jaxrs-clients:$version"
+  compile "com.palantir.remoting1:retrofit2-clients:$version"
+  compile "com.palantir.remoting1:dropwizard-servers:$version"
+  // optional support for PEM key store type using Bouncy Castle libraries:
+  //     compile "com.palantir.remoting1:pkcs1-reader-bouncy-castle:$version"
+  // optional support for PEM key store type using Sun libraries:
+  //     compile "com.palantir.remoting1:pkcs1-reader-sun:$version"
+}
+```
+
+
+## jaxrs-clients
+Provides the `JaxRsClient` factory for creating clients for JAX-RS services. Example:
+```java
+MyService service = JaxRsClient.builder()
+    .build(MyService.class, "my user agent", "https://my-server/");
+```
+The client is implemented using Feign; however, the Feign dependency is hidden away from both the Java API and the
+classpath (via shadowing).
+
+## retrofit2-clients
+Similar to `jaxrs-clients`, but generates clients using the Retrofit library. Example:
+
+```java
+MyService service = Retrofit2Client.builder()
+    .build(MyService.class, "my user agent", "https://my-server/");
+```
+
+## dropwizard-servers
+Provides Dropwizard/Jersey exception mappers for translating common JAX-RS exceptions to appropriate HTTP error codes. A
+Dropwizard server is configured for http-remoting as follows:
+
+```java
+public class MyServer extends Application<Configuration> {
+    @Override
+    public final void run(Configuration config, final Environment env) throws Exception {
+        env.jersey().register(new MyResource());
+        DropwizardServers.configure(env, config, MyServerc.class.getSimpleName(), false /** do not ship stacktraces */);
```

Comment:
should we change the boolean here to an enum so that comments aren't required?

     enum StackTraces {
         DO_NOT_SEND_OVER_WIRE,
         SEND_OVER_WIRE
     }